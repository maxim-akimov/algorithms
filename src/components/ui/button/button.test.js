import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';


import { Button } from './button';


describe('Тестирование компонента Circle', () => {
  it('Активная кнопка с текстом рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button type={"button"} title={"Текст кнопки"} disabled={false} isLoader={false}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Активная кнопка без текста рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button type={"button"} disabled={false} isLoader={false}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Заблокированная кнопка с текстом рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button type={"button"} title={"Текс кнопки"} disabled={true} isLoader={false}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Заблокированная кнопка без текста рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button type={"button"} disabled={true} isLoader={false}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Кнопка с индикацией загрузки с текстом рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button type={"button"} title={"Текс кнопки"} disabled={false} isLoader={true}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Кнопка с индикацией загрузки без текста рендерится без ошибок', () => {
    const tree = renderer
      .create(<Button type={"button"} disabled={false} isLoader={true}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });


  it('Клик на активную кнопку с текстом работает без ошибок', () => {
    window.alert = jest.fn();

    const onClick = () => {
      alert('click');
    }

    // Рендерим компонент
    render(<Button type={"button"} title={"Текст кнопки"} disabled={false} isLoader={false} onClick={onClick}/>)

    // Находим элемент ссылки
    const btn = screen.getByTitle('Текст кнопки');

    // Имитируем нажатие на ссылку
    fireEvent.click(btn);

    // Проверяем, что alert сработал с правильным текстом предупреждения
    expect(window.alert).toHaveBeenCalledWith('click');
  });
});