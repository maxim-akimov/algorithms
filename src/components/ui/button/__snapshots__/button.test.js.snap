// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование компонента Circle Активная кнопка без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={false}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Circle Активная кнопка с текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={false}
  title="Текст кнопки"
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Circle Заблокированная кнопка без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Circle Заблокированная кнопка с текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  title="Текс кнопки"
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Circle Кнопка с индикацией загрузки без текста рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование компонента Circle Кнопка с индикацией загрузки с текстом рендерится без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  title="Текс кнопки"
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
